homeassistant/components/unifiprotect/event.py

```diff
@@ -69,13 +162,24 @@ def _async_event_entities(
     data: ProtectData,
     ufp_device: ProtectAdoptableDeviceModel | None = None,
 ) -> list[ProtectDeviceEntity]:
+    # Define a mapping between the description keys and corresponding entity classes
+    entity_mapping: dict[str, type[ProtectDeviceEntity]] = {
+        "nfc": ProtectDeviceNFCEventEntity,
+        "fingerprint": ProtectDeviceFingerprintEventEntity,
+        "ring": ProtectDeviceRingEventEntity,
+    }
+
```

Comment:
You can add to the descriptions instead to avoid having this in two places

```diff
diff --git a/homeassistant/components/unifiprotect/event.py b/homeassistant/components/unifiprotect/event.py
index ffb87e734a3..6fd1b6fcf41 100644
--- a/homeassistant/components/unifiprotect/event.py
+++ b/homeassistant/components/unifiprotect/event.py
@@ -29,39 +29,9 @@ from .entity import EventEntityMixin, ProtectDeviceEntity, ProtectEventMixin
 class ProtectEventEntityDescription(ProtectEventMixin, EventEntityDescription):
     """Describes UniFi Protect event entity."""
 
+    cls: ProtectDeviceEntity | None = None
+
 
-EVENT_DESCRIPTIONS: tuple[ProtectEventEntityDescription, ...] = (
-    ProtectEventEntityDescription(
-        key="doorbell",
-        translation_key="doorbell",
-        device_class=EventDeviceClass.DOORBELL,
-        icon="mdi:bell",
-        ufp_required_field="feature_flags.is_doorbell",
-        ufp_event_obj="last_ring_event",
-        event_types=[EVENT_TYPE_DOORBELL_RING],
-    ),
-    ProtectEventEntityDescription(
-        key="nfc",
-        translation_key="nfc",
-        device_class=EventDeviceClass.DOORBELL,
-        icon="mdi:nfc",
-        ufp_required_field="feature_flags.support_nfc",
-        ufp_event_obj="last_nfc_card_scanned_event",
-        event_types=[EVENT_TYPE_NFC_SCANNED],
-    ),
-    ProtectEventEntityDescription(
-        key="fingerprint",
-        translation_key="fingerprint",
-        device_class=EventDeviceClass.DOORBELL,
-        icon="mdi:fingerprint",
-        ufp_required_field="feature_flags.has_fingerprint_sensor",
-        ufp_event_obj="last_fingerprint_identified_event",
-        event_types=[
-            EVENT_TYPE_FINGERPRINT_IDENTIFIED,
-            EVENT_TYPE_FINGERPRINT_NOT_IDENTIFIED,
-        ],
-    ),
-)
 
 
 class ProtectDeviceRingEventEntity(EventEntityMixin, ProtectDeviceEntity, EventEntity):
@@ -157,28 +127,55 @@ class ProtectDeviceFingerprintEventEntity(
             self.async_write_ha_state()
 
 
+EVENT_DESCRIPTIONS: tuple[ProtectEventEntityDescription, ...] = (
+    ProtectEventEntityDescription(
+        key="doorbell",
+        translation_key="doorbell",
+        device_class=EventDeviceClass.DOORBELL,
+        icon="mdi:bell",
+        ufp_required_field="feature_flags.is_doorbell",
+        ufp_event_obj="last_ring_event",
+        event_types=[EVENT_TYPE_DOORBELL_RING],
+        cls=ProtectDeviceRingEventEntity,
+    ),
+    ProtectEventEntityDescription(
+        key="nfc",
+        translation_key="nfc",
+        device_class=EventDeviceClass.DOORBELL,
+        icon="mdi:nfc",
+        ufp_required_field="feature_flags.support_nfc",
+        ufp_event_obj="last_nfc_card_scanned_event",
+        event_types=[EVENT_TYPE_NFC_SCANNED],
+        cls=ProtectDeviceNFCEventEntity,
+    ),
+    ProtectEventEntityDescription(
+        key="fingerprint",
+        translation_key="fingerprint",
+        device_class=EventDeviceClass.DOORBELL,
+        icon="mdi:fingerprint",
+        ufp_required_field="feature_flags.has_fingerprint_sensor",
+        ufp_event_obj="last_fingerprint_identified_event",
+        event_types=[
+            EVENT_TYPE_FINGERPRINT_IDENTIFIED,
+            EVENT_TYPE_FINGERPRINT_NOT_IDENTIFIED,
+        ],
+        cls=ProtectDeviceFingerprintEventEntity,
+    ),
+)
+
 @callback
 def _async_event_entities(
     data: ProtectData,
     ufp_device: ProtectAdoptableDeviceModel | None = None,
 ) -> list[ProtectDeviceEntity]:
     # Define a mapping between the description keys and corresponding entity classes
-    entity_mapping: dict[str, type[ProtectDeviceEntity]] = {
-        "nfc": ProtectDeviceNFCEventEntity,
-        "fingerprint": ProtectDeviceFingerprintEventEntity,
-        "ring": ProtectDeviceRingEventEntity,
-    }
-
     entities: list[ProtectDeviceEntity] = []
 
     for device in data.get_cameras() if ufp_device is None else [ufp_device]:
         for description in EVENT_DESCRIPTIONS:
             if not description.has_required(device):
                 continue
-
-            entity_class = entity_mapping.get(description.key)
-            if entity_class:
-                entities.append(entity_class(data, device, description))
+            entities.append(description.cls(data, device, description))
 
     return entities
 
```

Reply:
Note that typing might need to be adjusted